pygments/lexers/mcfunction.py

```diff
@@ -0,0 +1,272 @@
+"""
+    pygments.lexers.mcfunction
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexers for MCFunction and related languages.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, default, include, using
+from pygments.token import (Comment, Error, Keyword, Literal, Name, Number,
+                            Operator, Punctuation, String, Text, Token,
+                            Whitespace)
+
+
+__all__ = ['SNBTLexer', 'MCFunctionLexer']
+
+
+class SNBTLexer(RegexLexer):
+    """Lexer for stringified NBT, a data format used in Minecraft
+
+
+    .. versionadded:: ?
+    """
+
+    name = "SNBT"
+    url = ""
+    aliases = ["snbt"]
+    filenames = ["*.snbt"]
+    mimetype = ["text/snbt"]
+
+    tokens = {
+        "root": [
+            # We only look for the open bracket here since square bracket
+            #  is only valid in NBT pathing (which is a mcfunction idea).
+            (r"(?=\{)", Punctuation, "compound"),
```

Comment:
Hm, this isn't a default though (I can change Punctuation to Text like the normal Default), I only want the Lexer to continue if it finds a bracket. Or should i use `include("compound")` here? I don't want to lex the other aspects of a compound until we find the very first bracket. Maybe I'm misunderstanding.

Reply:
The way it is currently written, the lexer expects the file to start with a bracket. Otherwise, it will emit error tokens. I thought that brace was a requirement of the language while reading the code. If you want to start at the first brace, you should do something like

```
"root": [
    (r"[^\{]+", Text),
    (r"\{", Punctuation, "compound"),
] 
```

 By the way, can you add some tests for SNBT?